diff --git a/src/util/execute.go b/src/util/execute.go
index 03ac92d..9b7088c 100644
--- a/src/util/execute.go
+++ b/src/util/execute.go
@@ -9,6 +9,7 @@ import (
 
 	"github.com/hashicorp/go-multierror"
 	log "github.com/sirupsen/logrus"
+	"github.com/google/uuid"
 )
 
 const TimeoutExitCode = 124
@@ -36,11 +37,31 @@ func getErrorStr(err error, stderr *bytes2.Buffer) string {
 }
 
 func Execute(command string, args ...string) (stdout string, stderr string, exitCode int) {
+	isInstall := false
+	for _, arg := range args {
+		if strings.Contains(arg, "install") {
+			isInstall = true
+		}
+	}
+
+	for i := 0; i < len(args) - 1; i++ {
+		if args[i] == "--name" && args[i + 1] == "next-step-runner" {
+			args[i + 1] = fmt.Sprintf("next-step-runner-%s", uuid.NewString())
+		}
+	}
+
 	cmd := exec.Command(command, args...)
 	var stdoutBytes, stderrBytes bytes2.Buffer
 	cmd.Stdout = &stdoutBytes
 	cmd.Stderr = &stderrBytes
+
+	log.Infof("@@@@@@@@@@@@@@@@@@@@@@@@@@ %v %v %+v", isInstall, command, strings.Join(args, " ^ "))
+
 	err := cmd.Run()
+	if !isInstall {
+		// err = cmd.Run()
+	}
+
 	return stdoutBytes.String(), getErrorStr(err, &stderrBytes), getExitCode(err)
 }
 
